apps/src/code-studio/components/ShareAllowedDialog.jsx

```diff
@@ -323,227 +323,270 @@ class ShareAllowedDialog extends React.Component {
+              <a href="#" onClick={wrapShareClick(this.print, 'print')}>
+                <FontAwesome icon="print" style={{fontSize: 26}} />
+                <span>{i18n.print()}</span>
+              </a>
+            )}
+            {/* prevent buttons from overlapping when unpublish is pending */}
+            {this.isSocialShareAllowed() && !isUnpublishPending && (
+              <span>
+                {this.state.isFacebookAvailable && (
+                  <a
+                    href={facebookShareUrl}
+                    target="_blank"
+                    rel="noopener noreferrer"
+                    onClick={wrapShareClick(
+                      onClickPopup.bind(this),
+                      'SHARING_FB',
+                      this.props.appType
+                    )}
+                    style={styles.socialLink}
+                  >
+                    <FontAwesome icon="facebook" />
+                  </a>
                 )}
-                <div style={{clear: 'both'}}>
-                  <div style={styles.thumbnail}>
-                    <img
-                      style={styles.thumbnailImg}
-                      src={thumbnailUrl}
-                      alt={i18n.projectThumbnail()}
-                    />
-                  </div>
-                  <div>
-                    <Button
-                      color={Button.ButtonColor.brandSecondaryDefault}
-                      id="sharing-dialog-copy-button"
-                      icon="clipboard"
-                      style={{
-                        ...styles.button,
-                        ...styles.copyButton,
-                        ...(this.state.hasBeenCopied && styles.copyButtonLight),
-                      }}
-                      onClick={wrapShareClick(
-                        this.copy,
-                        'SHARING_LINK_COPY',
-                        this.props.appType
-                      )}
-                      text={i18n.copyLinkToProject()}
-                      value={shareUrl}
-                    />
-                    <DownloadReplayVideoButton
-                      style={{...styles.button, marginBottom: 8}}
-                      onError={this.replayVideoNotFound}
-                    />
-                  </div>
-                </div>
-                <div className="social-buttons" style={{marginTop: 12}}>
-                  <Button
-                    color={Button.ButtonColor.neutralDark}
-                    id="sharing-phone"
-                    href=""
+                {this.state.isTwitterAvailable && (
+                  <a
+                    href={twitterShareUrl}
+                    target="_blank"
+                    rel="noopener noreferrer"
                     onClick={wrapShareClick(
-                      this.showSendToPhone,
-                      'SHARING_LINK_SEND_TO_PHONE',
+                      onClickPopup.bind(this),
+                      'SHARING_TWITTER',
```

Comment:
Is there an actual `SHARING_TWITTER` constant that we can import in from somewhere instead of adding it as a string?